### PR TITLE
Modvend 66

### PR DESCRIPTION
**Pull request by komick**
API endpoints created for all models. Vendor_type added to vendor model. San-code moved from vendor_address into vendor model.

### CODE REVIEW

**Comment by dcrossleyau** [comment]
The apidocs configuration is adjusted.

Some of those errors seem to be due to examples, e.g. acq-models/mod-vendors/examples/agreement_get.sample is an empty file.

File: ramls/account.raml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+/account:
+  type:
+    collection:
+      exampleCollection: !include acq-models/mod-vendors/examples/account_collection.sample
+      exampleItem: !include acq-models/mod-vendors/examples/account_post.sample
+      schemaCollection: account_collection
+      schemaItem: account.json
+  get:
+    description: Get list of phone_numbers
```
Comment (craigmcnally) [inline ramls/account.raml]: fix description

File: ramls/account.raml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#%RAML 0.8
```
Comment (craigmcnally) [inline ramls/account.raml:1]: why not RAML 1.0?

File: ramls/interface.raml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+/interface:
+  type:
+    collection:
+      exampleCollection: !include acq-models/mod-vendors/examples/interface_collection.sample
+      exampleItem: !include acq-models/mod-vendors/examples/interface_post.sample
+      schemaCollection: interface_collection
+      schemaItem: interface.json
+  get:
+    description: Get list of EDIs
```
Comment (craigmcnally) [inline ramls/interface.raml]: fix description

File: ramls/vendor_address.raml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+/vendor_address:
+  type:
+    collection:
+      exampleCollection: !include acq-models/mod-vendors/examples/vendor_address_collection.sample
+      exampleItem: !include acq-models/mod-vendors/examples/vendor_address_post.sample
+      schemaCollection: vendor_address_collection
+      schemaItem: vendor_address.json
+  get:
+    description: Get list of EDIs
```
Comment (craigmcnally) [inline ramls/vendor_address.raml]: fix description

File: ramls/vendor_contact_person.raml [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+/vendor_contact_person:
+  type:
+    collection:
+      exampleCollection: !include acq-models/mod-vendors/examples/vendor_contact_person_collection.sample
+      exampleItem: !include acq-models/mod-vendors/examples/vendor_contact_person_post.sample
+      schemaCollection: vendor_contact_person_collection
+      schemaItem: vendor_contact_person.json
+  get:
+    description: Get list of EDIs
```
Comment (craigmcnally) [inline ramls/vendor_contact_person.raml]: fix description

File: ramls/vendor_email.raml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+/vendor_email:
+  type:
+    collection:
+      exampleCollection: !include acq-models/mod-vendors/examples/vendor_email_collection.sample
+      exampleItem: !include acq-models/mod-vendors/examples/vendor_email_post.sample
+      schemaCollection: vendor_email_collection
+      schemaItem: vendor_email.json
+  get:
+    description: Get list of EDIs
```
Comment (craigmcnally) [inline ramls/vendor_email.raml]: fix description

File: ramls/vendor_phone.raml [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+/vendor_phone:
+  type:
+    collection:
+      exampleCollection: !include acq-models/mod-vendors/examples/vendor_phone_collection.sample
+      exampleItem: !include acq-models/mod-vendors/examples/vendor_phone_post.sample
+      schemaCollection: vendor_phone_collection
+      schemaItem: vendor_phone.json
+  get:
+    description: Get list of EDIs
```
Comment (craigmcnally) [inline ramls/vendor_phone.raml]: fix description

File: ramls/vendor_type.raml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+/vendor_type:
+  type:
+    collection:
+      exampleCollection: !include acq-models/mod-vendors/examples/vendor_type_collection.sample
+      exampleItem: !include acq-models/mod-vendors/examples/vendor_type_post.sample
+      schemaCollection: vendor_type_collection
+      schemaItem: vendor_type.json
+  get:
+    description: Get list of EDIs
```
Comment (craigmcnally) [inline ramls/vendor_type.raml]: fix description

File: ramls/vendor_url.raml [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+/vendor_url:
+  type:
+    collection:
+      exampleCollection: !include acq-models/mod-vendors/examples/vendor_url_collection.sample
+      exampleItem: !include acq-models/mod-vendors/examples/vendor_url_post.sample
+      schemaCollection: vendor_url_collection
+      schemaItem: vendor_url.json
+  get:
+    description: Get list of EDIs
```
Comment (craigmcnally) [inline ramls/vendor_url.raml]: fix description

File: src/test/java/org/folio/rest/impl/VendorContactPersonTest.java [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.InputStream;
+
+import static com.jayway.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.*;
+
+@RunWith(VertxUnitRunner.class)
+public class VendorContactPersonTest {
```
Comment (craigmcnally) [inline src/test/java/org/folio/rest/impl/VendorContactPersonTest.java:32]: Many of these tests look very similar.  Consider pulling the common/boilerplate code out into a base test class like we did in mod-orders-storage.

File: ramls/account.raml (hunk 2 above)
Comment (komick) [reply ramls/account.raml:1]: This will be address in a future PR

File: src/test/java/org/folio/rest/impl/VendorContactPersonTest.java (hunk 10 above)
Comment (komick) [reply src/test/java/org/folio/rest/impl/VendorContactPersonTest.java:32]: Agreed.  Will resolve this in a future PR.

**Review by craigmcnally — APPROVED** [review]
I'm approving with the promise that the following will be addressed in separate stories:

* Migration to RAML 1.0/RMB 23
* Consolidation of APIs/models back into the main vendor record, except contact-related data

It would also be nice to see the unit tests refactored so the boilerplate/duplicate code is abstracted out into a base class like we've done for mod-orders-storage.

**Comment by komick** [comment]
> The apidocs configuration is adjusted.
> 
> Some of those errors seem to be due to examples, e.g. acq-models/mod-vendors/examples/agreement_get.sample is an empty file.
